### PR TITLE
#169258580 A user should be able to view  requests and a single request.

### PR DESCRIPTION
**Pull request by hezronkimutai**
#### What does this PR do?
- Implements requests table 
#### Description of Task to be completed?
This PR will enable the user to view trip requests:
- Created a reusable table ui component that will be used for all table-like components.
- Created a component for viewing a single request.
- Integrated the components with redux.
- Wrote tests
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend`
- `git checkout ft-implement-requests-table-169258580`
- Open your preferred browser and navigate to `/requests`
- To view a single request, click view button in the requests table.
#### What are the relevant pivotal tracker stories?
[#169258580](https://www.pivotaltracker.com/n/projects/2407428/stories/169258580)
#### Screenshots (if appropriate)
Screen Shot 2020-01-15 at 12 46 41
Screen Shot 2020-01-15 at 12 47 37

### CODE REVIEW

**Comment by NiyongaboEric** [comment]
Try to improve the description of PR. For example how to use it.

File: src/__tests__/views/trips/RequestsView.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { BrowserRouter as Router } from 'react-router-dom';
+import Requests from '../../../views/trips/RequestsView';
+
+
+let wrapper;
+const data = [{
+  id: 1, tripType: { tripType: 'one way' }, createdAt: '2011-30-24', status: { status: 'pending' },
+}];
```
Comment (victkarangwa) [inline src/__tests__/views/trips/RequestsView.test.js]: Would you mind placing this data in the mock file?

File: src/App.js [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
 import Signup from './containers/user/signup/signup';
 import VerifyEmail from './views/auth/VerifyEmail';
 import Signin from './containers/user/signin/signin';
 import Notfound from './NotFound';
+import Requests from './containers/Trips/Requests';
 import SocialLogin from './containers/user/social/SocialAuthLogin';
-import Dashboard from './views/Dashboard/index';
+import './assets/css/style.scss';
+
+// import Dashboard from './views/Dashboard/index';
```
Comment (NiyongaboEric) [inline src/App.js]: Remove comment.

File: src/__tests__/views/trips/CommonRequests.test copy.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  //   expect(plusDivs.mock.calls).toBeDefined();
+  // });
+  // it('it Should dispatch hit submit button successfully', async () => {
+  //   const plusDivs = jest.fn();
+  //   wrapper = shallow(
+  //     <Router>
+  //       <CommonSingleRequest entities={entities} navs={navs} tableHeads={tableHeads} />
+  //     </Router>,
+  //   );
+  //   const button = wrapper.find('#front');
```
Comment (NiyongaboEric) [inline src/__tests__/views/trips/CommonRequests.test copy.js]: Remove comment.

File: src/helpers/apiCall.js [hunk 4]
```diff
@@ -4,5 +4,5 @@ import dotenv from 'dotenv';
 dotenv.config();
 
 export default axios.create({
-  baseURL: process.env.BASE_URL,
+  baseURL: 'https://team-odd-bn-backend-staging.herokuapp.com/api/v1',
```
Comment (victkarangwa) [inline src/helpers/apiCall.js]: Can we have this link in env file instead?

File: src/views/Dashboard/sidebar/TopRightSide.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+
+  render() {
+    // const cc = new Image();
+
+    // cc.onerror = function () {
+    //   console.log('======hhhhhhh');
+    // };
+    // cc.src = this.state.image;
```
Comment (victkarangwa) [inline src/views/Dashboard/sidebar/TopRightSide.js]: Could you please get rid of these commented out lines?

File: src/views/trips/RequestsView.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  { className: 'tripIdNav', attribute: 'Trip Id', key: 'tripId' },
+  { className: 'tripTypeNav', attribute: 'Trip Type', key: 'tripType' },
+  { className: 'createdAtNav', attribute: 'Created At', key: 'createdAt' },
+  { className: 'status', attribute: 'Status', key: 'status' },
+  { className: 'actionNav', attribute: 'Action', key: 'action' },
+];
+const navs = [
+  { className: 'tableHeader', attribute: <h2>My Travel Requests</h2>, key: 'tableHeader' },
+  { className: 'search', attribute: <input type="text" placeholder="search by date" />, key: 'search' },
+];
```
Comment (victkarangwa) [inline src/views/trips/RequestsView.js]: I think both `tableHeads` and `navs` should be placed in the constants and then you import them.

**Review by NiyongaboEric — CHANGES_REQUESTED** [review]
So far so good @hezronkimutai. I would like to request the following
- I would like to see the view trip request online.
- I request we display a meaningful message to users if did not yet create it.
Screen Shot 2020-01-16 at 07 34 54 1

- I request to improve the available trip request route name. If users will see `URL/requests` How avail request URL for a manager will be? They are both reviewing all trips. I suggest either to be `URL/user/requests` or `URL/user-requests`

- I request you improve the description of PR so that we can test it locally.

Screen Shot 2020-01-16 at 08 11 32

**Review by victkarangwa — APPROVED** [review]
Good job @hezronkimutai , I think you are good to go since you implemented the feedback we left

**Review by NiyongaboEric — APPROVED** [review]

**Pull request merged**